app/(main)/account/recovery/page.tsx

```diff
@@ -0,0 +1,263 @@
+
+    useEffect(() => {
+      if (!userId || !secret || !expire) {
+        router.push('/login');
+        return;
+      }
+
+      setIsLoading(false);
+
+      if (isSignedIn) {
+        getUser();
+        router.push('/league/all');
+      }
+    }, [isSignedIn]);
+
+    return null;
+  };
+
+  const form = useForm<ResetUserPasswordSchemaType>({
+    resolver: zodResolver(ResetPasswordSchema),
+    defaultValues: {
+      password: '',
+      confirmPassword: '',
+    },
+  });
+
+  /**
+   * The current value of the 'password' field in the form.
+   * @type {string}
+   */
+  const password: string = useWatch({
+    control: form.control,
+    name: 'password',
+    defaultValue: '',
+  });
+
+  /**
+   * The current value of the 'confirmPassword' field in the form.
+   * @type {string}
+   */
+  const confirmPassword: string = useWatch({
+    control: form.control,
+    name: 'confirmPassword',
+    defaultValue: '',
+  });
+
+  /**
+   * Handles the form submission.
+   * @param data - The data from the form.
+   */
+  const onSubmit: SubmitHandler<ResetUserPasswordSchemaType> = async (data) => {
+    const { password } = data;
+    setIsSubmitting(true);
+
+    try {
+      await resetRecoveredPassword({
+        userId: userId as string,
+        token: secret as string,
+        password,
+      });
+
+      toast.custom(
+        <Alert
+          variant={AlertVariants.Success}
+          message="Your password has been reset successfully! Please login with your new password."
+        />,
+      );
+
+      setTimeout(() => router.push('/login'), 2000);
+    } catch (err) {
+      console.error('Error resetting password:', err);
+      let errorMessage = 'There was an error resetting your password.';
+
+      if (err instanceof Error) {
+        const customErr = err as CustomError;
+        if (customErr.type === 'user_invalid_token') {
+          errorMessage =
+            'Invalid recovery token. Your token may have expired or already been used.  Please try again.';
+        }
+      }
+
+      toast.custom(
+        <Alert variant={AlertVariants.Error} message={errorMessage} />,
+      );
+
+      setTimeout(() => router.push('/recover-password'), 2000);
+    } finally {
+      setIsSubmitting(false);
+    }
+  };
+
+  const isDisabled = !password || password !== confirmPassword;
+
+  return (
+    <section className="grid xl:grid-cols-2 xl:grid-rows-none">
+      <Suspense fallback={<GlobalSpinner />}>
+        <SearchParamsWrapper />
+      </Suspense>
+
+      <div className="row-span-1 grid w-full place-items-center from-[#4E160E] to-zinc-950 bg-gradient-to-b xl:h-screen xl:bg-gradient-to-b">
+        <Logo className="mx-auto w-52 xl:w-64 xl:place-self-end" src={logo} />
+        <div className="mx-auto grid gap-4 place-self-end px-8 pb-8 text-foreground">
+          <p className="hidden leading-7 xl:block">
+            Thank you... fantasy football draft, for letting me know that even
```

Comment:
Where is this quote from? We're not doing fantasy draft, it's football survivor.

Reply:
I don't know .. that is from the original home page on our main gridironsurvivor.com.